lucene/facet/src/java/org/apache/lucene/facet/sortedset/SortedSetDocValuesFacetCounts.java

```diff
@@ -414,4 +505,101 @@ public int compare(FacetResult a, FacetResult b) {
 
     return results;
   }
+
+  @Override
+  public List<FacetResult> getTopDims(int topNDims, int topNChildren) throws IOException {
+    // Creates priority queue to store top dimensions and sort by their aggregated values/hits and
+    // string values.
+    PriorityQueue<SortedSetDocValuesDimValueResult> pq =
+        new PriorityQueue<>(topNDims) {
+          @Override
+          protected boolean lessThan(
+              SortedSetDocValuesDimValueResult a, SortedSetDocValuesDimValueResult b) {
+            if (a.value.intValue() > b.value.intValue()) {
+              return false;
+            } else if (a.value.intValue() < b.value.intValue()) {
+              return true;
+            } else {
+              return a.dim.compareTo(b.dim) > 0;
+            }
```

Comment:
minor: Another option for doing this that some people prefer is below. I'm happy with either so don't feel you need to change this, but thought I'd suggest it in case you hadn't seen this approach:

```suggestion
            int cmp = Integer.compare(a.value.intValue(), b.value.intValue());
            if (cmp == 0) {
              cmp = b.dim.compareTo(a.dim);
            }
            return cmp < 0;
```